### DISCUSSION TITLE
Inconsistent APIRouter constructor signature

### DISCUSSION THREAD

**Discussion by dorinclisu** [fastapi/fastapi#6940] [Questions] [answered]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.

### Environment
* OS: Linux
* FastAPI Version: 0.62.0
* Python: 3.9
* Mypy: 0.790

### Description
```Python
from fastapi import APIRouter
router = APIRouter(responses={})
```
`mypy(error): Unexpected keyword argument "responses" for "APIRouter"`

Despite the mypy error, it runs as expected. The question is why `responses` is not included in the signature.
This is what VSCode shows me as constructor signature:
```
APIRouter(routes: Optional[List[routing.BaseRoute]]=None, redirect_slashes: bool=True,
  default: Optional[ASGIApp]=None, dependency_overrides_provider: Optional[Any]=None,
  route_class: Type[APIRoute]=APIRoute, default_response_class: Optional[Type[Response]]=None,
  on_startup: Optional[Sequence[Callable]]=None, on_shutdown: Optional[Sequence[Callable]]=None)
```

**Comment by falkben** [comment]
I cannot replicate this error.

How are you running mypy?

For me, the signature shows `responses` (see below) and I don't see a mypy error either

![image](https://user-images.githubusercontent.com/653031/103446032-08451200-4c49-11eb-802e-dee859a7b57e.png)

**Comment by dorinclisu** [comment, marked as answer]
My fault, I am running 0.62.0 in Docker container but VSCode + mypy was "seeing" 0.61.2 in host OS, which did not support responses in APIRouter.

So the problem was FastAPI version mismatch between host and Docker container.

**Comment by dorinclisu** [comment]
I think the only way to prevent such issues is to install the Remote - Containers vscode extension so that mypy always runs against the packages inside the container. Until now I've used plain volume mounts which was ok for the editing itself.

**Comment by tiangolo** [comment]
Thanks for the help here @falkben ! 👏  🙇 

Thanks for reporting back and closing the issue @dorinclisu  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
